src/libfsm/subgraph.c

```diff
@@ -182,3 +182,125 @@ fsm_state_duplicatesubgraphx(struct fsm *fsm, fsm_state_t state,
 	return 1;
 }
 
+void
+fsm_subgraph_capture_start(struct fsm *fsm, struct fsm_subgraph_capture *capture)
+{
+	assert(fsm != NULL);
+	assert(capture != NULL);
+
+	capture->start = fsm_countstates(fsm);
+}
+
+void
+fsm_subgraph_capture_stop(struct fsm *fsm, struct fsm_subgraph_capture *capture)
+{
+	assert(fsm != NULL);
+	assert(capture != NULL);
+
+	capture->end = fsm_countstates(fsm);
+}
```

Comment:
This is a neat idea, at least while we have suitable backing storage to keep it efficient. I like it.